include/deal.II/base/partitioner.templates.h

```diff
@@ -633,70 +607,90 @@ namespace Utilities
 #    else
           if (vector_operation == dealii::VectorOperation::add)
             {
-              for (const auto &import_range : import_indices_data)
+              const unsigned int import_indices_plain_dev_size =
+                import_indices_plain_dev.size();
+              for (unsigned int i = 0; i < import_indices_plain_dev_size; ++i)
```

Comment:
It seems that you can still use a range-based for loop here and I expect `clang-tidy` to complain if it sees this loop.